pkgs/applications/editors/emacs/elisp-packages/tsc/default.nix

```diff
@@ -0,0 +1,74 @@
+{ lib
+, symlinkJoin
+, melpaBuild
+, fetchFromGitHub
+, rustPlatform
+, writeText
+, clang
+, llvmPackages
+}:
+with lib.licenses;
+with rustPlatform;
+with llvmPackages;
+
+let
+  version = "0.16.1";
+
+  meta = {
+    description = "The core APIs of the Emacs binding for tree-sitter.";
+    license = mit;
+    maintainers = with maintainers; [ pimeys ];
+  };
+
+  src = fetchFromGitHub {
+    owner = "emacs-tree-sitter";
+    repo = "elisp-tree-sitter";
+    rev = version;
+    sha256 = "sha256-tAohHdAsy/HTFFPSNOo0UyrdolH8h0KF2ekFXuLltBE=";
+  };
+
+  tsc = melpaBuild rec {
+    inherit src;
+    inherit version;
+
+    pname = "tsc";
+    commit = version;
+
+    sourceRoot = "source/core";
+
+    recipe = writeText "recipe" ''
+      (tsc
+      :repo "emacs-tree-sitter/elisp-tree-sitter"
+      :fetcher github)
+    '';
+  };
+
+  tsc-dyn = buildRustPackage {
+    inherit version;
+    inherit src;
+
+    pname = "tsc-dyn";
+    commit = version;
```

Comment:
I might be mistaken but I don't think `buildRustPackage` uses a `commit` arg? (here is the [definition](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/rust/build-rust-package/default.nix))